src/raw.rs

```diff
@@ -631,7 +620,7 @@ impl Location {
     #[inline]
     fn of_raw(index: usize) -> Location {
         // Calculate the bucket index based on ⌊log2(index)⌋.
-        let bucket = BUCKETS - ((index + 1).leading_zeros() as usize) - 1;
```

Comment:
Hmm, why this change?

Reply:
Because the `- 1` has been moved to the `BUCKETS` constant, but we know this formula should be pointwise equal to what it was before.

More satisfyingly, the correct value of `bucket` is `⌊log₂(index + 1)⌋ − ZERO_BUCKET`. Observe that `⌊log₂(index + 1)⌋ = usize::BITS - 1 - (index + 1).leading_zeros()` ([ref](https://doc.rust-lang.org/1.85.0/src/core/num/nonzero.rs.html#1492)). Then, previously we had:

```
⌊log2(index + 1)⌋ − ZERO_BUCKET
= usize::BITS - 1 - (index + 1).leading_zeros() - ZERO_BUCKET
= usize::BITS - ZERO_BUCKET - 1 - (index + 1).leading_zeros()
= BUCKETS - (index + 1).leading_zeros() - 1
```

But now we have:

```
⌊log2(index + 1)⌋ − ZERO_BUCKET
= usize::BITS - ZERO_BUCKET - 1 - (index + 1).leading_zeros()
= BUCKETS - (index + 1).leading_zeros()
```